### ISSUE TITLE
Allow capture of console messages from Daedalus ITN namespaceHelper - and make them more human readable

### ISSUE THREAD

**Issue by josephj11** [input-output-hk/daedalus#1880]
Until today's version (which is dramatically better!), Daedalus was giving me a lot of grief, so I run it in a Linux terminal. When I do, I see a bunch of informative messages that are very hard to read. Maybe they're json or something similar.

Originally, I was able to capture these and pipe them through a filter to make them more readable, but then you changed the namespaceHelper script and broke that.

I am attaching my modified namespaceHelper2 script, itn, the script that calls it, and itn_strip3, the readability filter. (I added .txt extensions to them so they would be accepted uploads.)
[itn_strip3.txt](https://github.com/input-output-hk/daedalus/files/4295690/itn_strip3.txt)
[itn.txt](https://github.com/input-output-hk/daedalus/files/4295691/itn.txt)
[namespaceHelper2.txt](https://github.com/input-output-hk/daedalus/files/4295692/namespaceHelper2.txt)

What my changes do:
Open a new terminal and run Daedalus in it
Pipe the output from stdout and stderr into a filter script
The filter script looks at each line of output
If the line contains a "msg" field, it prints the timestamp converted to local time and the message text
If there is no "msg" field, it prints the line as is.

This is very helpful because it makes it readable and lets me know how long it has been sitting doing nothing visible when it hasn't admitted to starting to sync, etc.

What I would like:

Change namespaceHelper so it doesn't use exec - so I can pipe the output into my script without having to manually modify it myself.

Less important, but useful: Change the console output similar to what my filter does, so it's more human-readable.

Since my scripts are a hack, I didn't try to make them generic so they'd work on other systems, but I don't think it wouldn't take much to fix that. They're all GPL 2 or greater, so feel free to use them if you wish.

I just submitted another support request #1879  about some strange messages Daedalus is emitting every time it runs. The reason I was able to notice this is because my filter makes it humanly visible.

Something like this may also come in handy for future versions if there are any issues these messages can reveal.

**Comment by nikolaglumac** [comment]
Hi @josephj11,
`namespaceHelper`, along with the Daedalus frontend logging formatter (https://github.com/input-output-hk/daedalus/blob/v2-integration/source/common/utils/logging.js#L67) are following IOHK structured logging specification. The output is indeed not human-readable but that was never an intention as the logs are mostly used as input into log analyzing systems we use internally.

**Comment by josephj11** [comment]
I figured as much. However, I am pointing out the benefit of making them human-readable for bug reporting ... And, in my case, I can do that myself - if you just get rid of the exec which prevents piping the the output into a filter such as the one I am using.

I have no ITN user demographics, but I would bet that the average ITN user is a lot more than just an investor and would be capable of putting such additional information to good use.

In any case, it's not an either/or question. You can still log the information anywhere you want in its original format (as my script also does.) Providing user readable output is very easy and low cost - my script is only a few lines of simple code and already has tangible benefits.

If you were to provide something similar, I assume it would be more sophisticated because you know what all those fields actually mean.

The bottom line, however is that I just want you to stop preventing me from using my filter. I would be quite satisfied if you just get rid of the exec in your script so I don't have to do it manually.

Since it works fine in my edited version without the exec, I'm not sure why you would have added it in the first place. (It wasn't there originally.) Is there some benefit to using exec that I don't see?

Even in your script, there is no further code beyond the one if statement, so the use of exec provides almost exactly zero performance gain. All it does is allow the one process to exit immediately instead of waiting until Daedalus terminates - and in Linux, there are so many processes running anyway so having one more idle one makes no difference at all.

**Comment by cleverca22** [comment]
the `exec` in the shell script should cause the child to inherit the same pid, and stdout/stderr will remain unchanged and obey any redirections you already put in place

**Comment by nikolaglumac** [comment]
@josephj11 is this still relevant? There is no more Daedalus ITN...

**Comment by josephj11** [comment]
I haven't tried it with the Shelley version so I'll close this. Thanks.
